File: src/main/java/me/nylestroke/nylemod/painting/ModPaintings.java
package me.nylestroke.nylemod.painting;

import me.nylestroke.nylemod.NylemodExample;
import net.minecraft.entity.decoration.painting.PaintingMotive;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class ModPaintings {

    public static final PaintingMotive MARATHON = registerPainting("marathon", new PaintingMotive(16, 16));
    public static final PaintingMotive FAMILY = registerPainting("family", new PaintingMotive(16, 32));

    private static PaintingMotive registerPainting(String name, PaintingMotive paintingMotive) {
        return Registry.register(Registry.PAINTING_MOTIVE, new Identifier(NylemodExample.MOD_ID, name), paintingMotive);
    }


    public static void registerPaintings() {
        NylemodExample.LOGGER.info("Registering Paintings for " + NylemodExample.MOD_ID);
    }

}

File: src/main/kotlin/me/nylestroke/nylemod/painting/ModPaintings.kt
package me.nylestroke.nylemod.painting

import me.nylestroke.nylemod.NylemodExample
import net.minecraft.entity.decoration.painting.PaintingVariant
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry

object ModPaintings {
    val MARATHON: PaintingVariant = registerPainting("marathon", PaintingVariant(16, 16))
    val FAMILY: PaintingVariant = registerPainting("family", PaintingVariant(16, 32))
    private fun registerPainting(name: String, paintingMotive: PaintingVariant): PaintingVariant {
        return Registry.register<PaintingVariant, PaintingVariant>(Registry.PAINTING_VARIANT, Identifier(NylemodExample.MOD_ID, name), paintingMotive)
    }

    fun registerPaintings() {
        NylemodExample.LOGGER.info("Registering Paintings for " + NylemodExample.MOD_ID)
    }
}